src/NinjaTools.FluentMockServer.API/Proxy/Visitors/ComparasionVisitor.cs

```diff
@@ -0,0 +1,169 @@
+            using var reader = new StreamReader(HttpRequest.Body);
+            var httpBody = reader.ReadToEnd();
+
+            if (visitable.MatchExact.HasValue && visitable.MatchExact.Value  && httpBody == visitable.Content)
+                return Pass();
+
+            return Fail();
+        }
+
+        /// <inheritdoc />
+        public double Visit(HeaderCollection? visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrEmpty(visitable.Headers, nameof(visitable) + nameof(HeaderCollection));
+
+            if (HttpRequest.Headers is null && visitable.Headers.EnsureNotNullNotEmpty() is { })
+                return Fail();
+
+            var success = visitable.Headers.Where(v =>
+            {
+                if (!HttpContext.Request.Headers.TryGetValue(v.Key, out var value))
+                    return false;
+
+                return !value.Except(v.Value).Any();
+
+            }).Any();
+
+            return success ? Pass() : Fail();
+        }
+
+
+        /// <inheritdoc />
+        public double Visit(CookieCollection? visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrEmpty(visitable, nameof(visitable) + nameof(CookieCollection));
+
+            if (HttpRequest.Cookies is null && visitable is { } v && v.Any())
+                Fail();
+
+            if (visitable.Except(HttpContext.Request.Cookies).EnsureNotNullNotEmpty().Any())
+                return Fail();
+
+            return Pass();
+        }
+
+        /// <inheritdoc />
+        public double Visit(QueryCollectionWrapper visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrEmpty(visitable, nameof(visitable) + nameof(QueryCollectionWrapper));
+            if (!HttpRequest.QueryString.HasValue && visitable.QueryString.HasValue)
+                return Fail();
+
+            if (visitable.Query == HttpRequest.QueryString.Value)
+                return Pass();
+
+            return Fail();
+        }
+
+        /// <inheritdoc />
+        public double Visit(PathCollection? visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrWhiteSpace(visitable, nameof(visitable) + nameof(PathCollection));
+
+            if (!HttpRequest.Path.HasValue && visitable is { } v && v.PathString.HasValue)
+                return  Fail();
+
+            if (visitable.ToPath() == HttpRequest.Path.Value)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802644)